micrometer-core/src/main/java/io/micrometer/core/instrument/simple/SimpleMeterRegistry.java

```diff
@@ -179,25 +180,51 @@ public String getMetersAsString() {
     }
 
     private String toString(Meter meter) {
-        String name = meter.getId().getName();
-        Meter.Type type = meter.getId().getType();
-        String baseUnit = meter.getId().getBaseUnit();
-        String tags = meter.getId().getTags().stream()
+        Meter.Id id = meter.getId();
+        String tags = id.getTags().stream()
                 .map(this::toString)
                 .collect(Collectors.joining(", "));
-
-        return StreamSupport.stream(meter.measure().spliterator(), false)
-                .map(measurement -> toString(name, type, baseUnit, tags, measurement))
-                .collect(Collectors.joining("\n"));
+        String baseUnit = id.getBaseUnit();
+        String meterUnitSuffix = baseUnit != null ? " " + baseUnit : "";
+        String measurements = StreamSupport.stream(meter.measure().spliterator(), false)
+                .map((measurement) -> toString(measurement, meter, meterUnitSuffix))
+                .collect(Collectors.joining(", "));
+        return String.format("%s(%s)[%s]; %s", id.getName(), id.getType(), tags, measurements);
     }
 
     private String toString(Tag tag) {
         return String.format("%s='%s'", tag.getKey(), tag.getValue());
     }
 
-    private String toString(String name, Meter.Type type, @Nullable String baseUnit, String tags, Measurement measurement) {
-        String statistic = measurement.getStatistic().toString().toLowerCase();
-        String unit = baseUnit != null ? "." + baseUnit : "";
-        return String.format("%s.%s%s(%s)[%s] %s", name, statistic, unit, type, tags, measurement.getValue());
+    private String toString(Measurement measurement, Meter meter, String meterUnitSuffix) {
+        Statistic statistic = measurement.getStatistic();
+        return String.format("%s=%s%s",
+                statistic.toString().toLowerCase(),
+                measurement.getValue(),
+                getUnitSuffix(statistic, meter, meterUnitSuffix));
+    }
+
+    private String getUnitSuffix(Statistic statistic, Meter meter, String meterUnitSuffix) {
+        switch (statistic) {
+            case DURATION:
+            case TOTAL_TIME:
+                return meterUnitSuffix;
+
+            case MAX:
+                if (meter instanceof Timer) {
```

Comment:
Should we add `DistributionSummary` here too?
Or do we need the `instaceof` check at all since if there is a max of something that has a unit we probably what to show what that unit was?